controllers/storagecluster/cephconfig.go

```diff
@@ -145,11 +164,17 @@ func getRookCephConfig(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (s
 			return "", fmt.Errorf("no CIDR is detected")
 		}
 		cidrName := strings.Join(cidrNameArray, ",")
-		rookConfigData, err := updateRookConfig(defaultRookConfigData, globalSectionKey, publicNetworkKey, cidrName)
+		rookConfigData, err = updateRookConfig(rookConfigData, globalSectionKey, publicNetworkKey, cidrName)
 		if err != nil {
 			return "", fmt.Errorf("failed to set network configuration for rook: %v", err)
 		}
-		return rookConfigData, nil
 	}
-	return defaultRookConfigData, nil
+	// additonal rook configs when it is a provider cluster
+	if sc.Spec.AllowRemoteStorageConsumers {
+		rookConfigData, err = updateRookConfigFromMap(rookConfigData, globalSectionKey, providerSCRookConfigs)
```

Comment:
You could use `updateRookConfig` directly and avoid this new wrapper.

Reply:
Changed implementation to fetch and merge configs using ini library